### PR TITLE
Ads fixes for Android Ads Custom Notifications

### PR DESCRIPTION
**Pull request by yachtcaptain23**
swipe_up_ad

Closes https://github.com/brave/brave-browser/issues/13038

## Submitter Checklist:

- [X] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [X] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [X] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [X] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [X] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [X] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
On Android, go through OOBE and observe the new push notification test. Try swiping up to dismiss and make sure tapping opens up the notification.

### CODE REVIEW

**Review by tmancey — APPROVED** [review]
LGTM

**Comment by yachtcaptain23** [comment]
android-dismiss-alpha

File: browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java [hunk 1]
```diff
@@ -70,44 +81,73 @@ public static void showAdNotification(Context context, final String notification
         window.setFlags(WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL,
```
Comment (samartnik) [inline browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java:81]: Do we still need this call given that flag had been set before?

File: browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java [hunk 2]
```diff
@@ -70,44 +81,73 @@ public static void showAdNotification(Context context, final String notification
...
+        mNotificationId = notificationId;
+        mAdsDialog.findViewById(R.id.brave_ads_custom_notification_popup).setOnTouchListener(new View.OnTouchListener() {
             @Override
-            public void onClick(View view) {
-                mAdsDialog.dismiss();
-                mAdsDialog = null;
-                BraveAdsNativeHelper.nativeOnCloseAdNotification(
-                        Profile.getLastUsedRegularProfile(), mNotificationId, true);
-                mNotificationId = null;
+            public boolean onTouch(View v, MotionEvent event) {
```
Comment (samartnik) [inline browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java:93]: Are we handling swipes up and down? Usually notifications are dismissed by swiping left and right on Android.

File: browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java [hunk 3]
```diff
@@ -70,44 +81,73 @@ public static void showAdNotification(Context context, final String notification
...
+                float deltaYDp;
+                float y;
+                switch (event.getAction()) {
+                    case MotionEvent.ACTION_DOWN:
+                        mYDown = v.getY() - event.getRawY();
+                        break;
+                    case MotionEvent.ACTION_MOVE:
+                        deltaY = event.getRawY() + mYDown;
+                        if (deltaY > 0) {
+                          deltaY = 0;
```
Comment (samartnik) [inline browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java:104]: nit: formatting

File: browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java (hunk 1 above)
Comment (yachtcaptain23) [reply browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java:81]: Will fix.

File: browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java (hunk 2 above)
Comment (yachtcaptain23) [reply browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java:93]: Ads team wanted to copy the behavior from iOS. @tmancey would you chime in? I don't know if I can make it in time for trains, but we can let it sit and if users want left/right swipe I can do so.

File: browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java (hunk 3 above)
Comment (yachtcaptain23) [reply browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java]: Fixed.

File: browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java (hunk 2 above)
Comment (tmancey) [reply browser/brave_ads/android/java/org/chromium/chrome/browser/dialogs/BraveAdsNotificationDialog.java:90]: @jenn-rhim Your thoughts? For me I like to follow the systems default, which as above is left/right

**Comment by tmancey** [comment]
@yachtcaptain23 Can you please run `npm run lint` as linter is failing, thanks

**Review by samartnik — APPROVED** [review]
lgtm, the only concern if the direction of swiping

**Comment by tmancey** [comment]
@samartnik Discussed with @jenn-rhim and swipe up is preferred. Thanks

**Review by tmancey — APPROVED** [review]
LGTM

**Comment by tmancey** [comment]
CI failed for unrelated JUnit tests

**Pull request merged**